windrose/windrose_main.py

```diff
@@ -0,0 +1,123 @@
+import os
+import pandas as pd
+import numpy as np
+from windrose import WindroseAnalysis
+import time
+from pathlib import Path
+import argparse
+
+
+class WindroseProcessor:
+    def __init__(self, root_directory, directory='windRoseData'):
+        self.root_directory = Path(root_directory)
+        self.directory = Path(directory)
+
+        if not self.directory.exists():
+            self.directory.mkdir(parents=True)
+
+    def process_site_files(self):
+        site_list = [
+            d for d in self.root_directory.iterdir()
+            if d.is_dir()
+        ]
+        # print(site_list)
+
+        for site in site_list:
+            file_list = os.listdir(site)
+            filtered_files = [
+                file for file in file_list
+                if "_BASE_" in file and "_HH_" in file
+            ]
+            # print(site)
+
+            for filtered_file in filtered_files:
+                file_path = site / filtered_file
+                df = pd.read_csv(file_path, skiprows=[0, 1])
+
+                if 'WS' not in df.columns or 'WD' not in df.columns:
+                    print("Both 'WS' and 'WD' or either columns are missing.")
+                else:
+                    self.process_season_data(df, site)
+
+    def process_season_data(self, df, site):
+        df_1 = df[(df['WD'] != -9999) | (df['WS'] != -9999)]
+        speed = df_1['WS']
+        direction = df_1['WD']
+        timestamp = df_1['TIMESTAMP_START']
+        time_series = pd.to_datetime(timestamp, format='%Y%m%d%H%M')
+        hourz = time_series.dt.strftime('%H%M').astype(int)
+        monthz = time_series.dt.month
+        year_start = time_series.dt.year.iloc[0]
+        year_end = time_series.dt.year.iloc[-1]
+
+        all_season_df = pd.DataFrame({
+            'month': monthz,
+            'hour': hourz,
+            'speed': speed,
+            'direction': direction
+        })
+
+        all_season = list(range(1, 13))
+        spring = [3, 4, 5]
+        summer = [6, 7, 8]
+        autumn = [9, 10, 11]
+        winter = [12, 1, 2]
+        day_time = list(range(600, 1810, 10))
+        night_time = list(range(0, 610, 10)) + list(range(1800, 2410, 10))
+        seasons = [all_season, spring, summer, autumn, winter]
+        day_night = [day_time, night_time]
+        season_names = ['Full_Year', 'Spring', 'Summer', 'Autumn', 'Winter']
+        time_names = ['Daytime', 'Nighttime']
+        ndirections = 16
+        bins1 = [0, 1, 4, 9, 16, 26, 37, 50]
+        percentile95 = np.percentile(speed, 95)
+        bins2 = np.linspace(0, percentile95, 8)
+        vwinds_list = [bins1, bins2]
```

Comment:
What does `bins1` and `bins2` refer to? It is not clear by looking at the variable names

Reply:
bins1 represents a predefined set of wind speed intervals. This interval might be based on standard meteorological classifications. For example, [0, 1, 4, 9, 16, 26, 37, 50] might correspond to various wind speed categories like calm, light breeze, moderate wind, etc.

bins2 represents a set of wind speed intervals calculated based on the dataset itself. The percentile95 variable is used to calculate the 95th percentile of wind speeds, and bins2 is created by dividing the range from 0 to this 95th percentile into equal intervals.